### DISCUSSION TITLE
How to implement typical behavior of process suspension via `Ctrl+Z` and subsequent sending to background via `bg`?

### DISCUSSION THREAD

**Discussion by whatbh** [leo-arch/clifm#274] [Q&A]
In Bash, a foreground process can be suspended using `Ctrl+Z`, the terminal sends a signal to suspend the process **and returns to the shell prompt**. The suspended process can then be resumed in background or foreground by using `bg` or `fg` command, respectively.

In `clifm`, pressing `Ctrl+Z` can somewhat suspend the process but it **never returns to the shell prompt**. The shell now is in an unusable state, even if the suspended process is killed, nothing works, `Ctrl+C`, `Ctrl+D` does not work.

On the other hand, I don't want to sacrifice the ability to place jobs on-the-fly by having to specifically run a job with an ampersand.

Is there anyway to do this but I have overlooked  somewhere? Or this feature is not implemented in `clifm` yet? Should I open an issue regarding this matter?

Thanks.

**Comment by leo-arch** [comment]
Hi @whatbh. Thanks for your comments.

> In Bash, a foreground process can be suspended using Ctrl+Z, the terminal sends a signal to suspend the process and returns to the shell prompt. The suspended process can then be resumed in background or foreground by using bg or fg command, respectively.

Job control, beyond the basic run-in-the-background via ampersand, is not implemented in **clifm**. This is a typical shell feature, true, but **clifm** is not a shell, but a file manager. However, it might be implemented in the future. Please open an issue requesting this feature, so that we can keep track of it.

> In clifm, pressing Ctrl+Z can somewhat suspend the process but it never returns to the shell prompt. The shell now is in an unusable state, even if the suspended process is killed, nothing works, Ctrl+C, Ctrl+D does not work.

This is a bug. Please open a second issue and write a bug report.

**Reply by whatbh** [reply]
Excellent, thank you for clarifying that! Appreciate all the effort I will proceed to open an issue regarding this matter.
